### PR TITLE
Rename infra environments to host inventory

### PR DESCRIPTION
**Pull request by rawagner**
as described in the doc https://docs.google.com/document/d/1WGGjqC_fzrWf1cUigu8pa6hBwcf__3tW5R-UHqyhrms , we want to update the naming as the old one is too confusing

### CODE REVIEW

File: frontend/src/routes/Infrastructure/InfraEnvironments/HostInventoryBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import { AcmButton } from '@stolostron/ui-components'
+import { ActionGroup, ButtonVariant, Card, CardBody, CardFooter, CardTitle, StackItem } from '@patternfly/react-core'
+import { useState } from 'react'
+import { ExternalLinkAltIcon } from '@patternfly/react-icons'
+import { useTranslation } from '../../../lib/acm-i18next'
+import { DOC_LINKS } from '../../../lib/doc-util'
+
+const localStorageKey = `HostInventoryBannerDismissed`
+// t('Hosts are discovered and stored in infrastructure environments. You can manage your host inventory with infrastructure environments which allow to set the same configurations for all hosts that it contains.')
```
Comment (mareklibra) [inline frontend/src/routes/Infrastructure/InfraEnvironments/HostInventoryBanner.tsx:11]: nit: leftover

File: frontend/public/locales/en/translation.json [hunk 2]
```diff
@@ -1517,6 +1520,7 @@
     "managedClusterSet.form.removed": "Removed",
     "managedClusterSet.form.transferred": "Transferred",
     "managedClusterSet.form.unchanged": "No change",
+    "Managing hosts with infrastructure environments makes it easy to set settings across multiple hosts.": "Managing hosts with infrastructure environments makes it easy to set settings across multiple hosts.",
```
Comment (mareklibra) [inline frontend/public/locales/en/translation.json:1523]: opinion: I understand the benefits of having "real texts" as keys for translations but I would shorten them for similarly long cases

Comment (rawagner) [reply frontend/public/locales/en/translation.json:1523]: I disagree.
How do we say which key is too long ?
Also it does not really matter how this file looks like, its not something we would be looking at. The important part, which needs to be readable, is in the actual code in which case a using a real text is a clear winner over some key

**Review by mareklibra — DISMISSED** [review]

**Pull request merged**